java/core/src/main/java/com/microsoft/bond/ListBondType.java

```diff
@@ -154,7 +154,7 @@ protected final void serializeField(
             StructBondType.StructField<List<TElement>> field) throws IOException {
         this.verifySerializedNonNullableFieldIsNotSetToNull(value, field);
         int count = value.size();
-        if (count == 0 && field.isOptional()) {
+        if (!field.isDefaultNothing() && count == 0 && field.isOptional()) {
             context.writer.writeFieldOmitted(BondDataType.BT_LIST, field.getId(), field);
         } else {
```

Comment:
Let's say the field is `= nothing` and hasn't been set. Where does the writing of the field get skipped completely? In the generated `BOND_TYPE` serialize method? If it doesn't get skipped at a higher level, this condition looks like it will go down the `writeFieldBegin` path, which doesn't seem right: there's no value to write...